apps/todomvc/todomvc-web-components/dist/utils/script-loader.js

```diff
@@ -0,0 +1,62 @@
+import { requestIdleCallback } from "./request-idle-callback.js";
+
+function createScript({type}) {
+    return new Promise((resolve) => {
+        const scriptEl = document.createElement('script');
+        if (type !== "") scriptEl.type = type;
+        return resolve({ success: true, type: "createScript", scriptEl});
+    });
+}
+
+function addScript({scriptEl, location}) {
+    return new Promise((resolve) => {
+        if (location === 'head') {
+            document.head.appendChild(scriptEl);
+        } else {
+            document.body.appendChild(scriptEl);
+        }
+        return resolve({ success: true, type: "addScript"});
+    });
+}
```

Comment:
Are there plans to make this async at some point? If not, then I'm not sure we need the Promise here, since all the operations are sync (here and `createScript`, `buildScript`)

Reply:
reworked it and only kept promises for the loading